### ISSUE TITLE
PCA Quality

### ISSUE THREAD

**Issue by GeorgePearse** [tensorflow/tensorboard#5403]
If I try to run PCA before loading data into Tensorboard to improve loading speeds (with Sklearn) I get results that are visually much much worse. Any idea why this might be?

**Comment by ericdnielsen** [comment]
How are you running your PCA?  Is the process overwriting the values stored in the event files?

**Comment by GeorgePearse** [comment]
Instead of feeding the SummaryWriter raw embeddings (2688 dims), I was trying to run the PCA ahead of time to reduce the dimensionality to 3. Figured this would be much more efficient and allow the embedding projector to act as a great 3d scatter graph with interactive metadata etc, but the clustering almost completely fails when I go through this process with sklearn's PCA. I was expecting the results to be almost identical. I though PCA always achieved an exact result, just with different runtimes, but think I've just realised that it's normally some approximation. Cheers!
